File: frontend/src/types/compliance.ts
export type ComplianceSource = "wazuh" | "ssh";

export interface ComplianceControl {
  control_id: string;
  title: string;
  description: string;
  result: "passed" | "failed" | "not_applicable";
  risk_level: "critical" | "high" | "medium" | "low";
  remediation: string;
  rationale?: string;
  references?: string;
}

export interface ComplianceRecommendation {
  priority: number;
  title: string;
  description: string;
  remediation_steps: string;
}

export interface ComplianceReportSummary {
  id: string;
  tenant_id: string;
  server_id: string;
  source: ComplianceSource;
  policy_name: string;
  score_pct: number;
  total_checks: number;
  passed: number;
  failed: number;
  not_applicable: number;
  created_at: string;
}

export interface ComplianceReport extends ComplianceReportSummary {
  agent_id: string | null;
  policy_id: string | null;
  controls: ComplianceControl[];
  ai_summary: string;
  ai_recommendations: ComplianceRecommendation[];
}

export interface ComplianceGenerateRequest {
  server_id: string;
  policy_id?: string;
  force_source?: "wazuh" | "ssh";
}

File: backend/app/schemas/compliance.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class ComplianceReportRead(BaseModel):
    id: UUID
    tenant_id: UUID
    server_id: UUID
    source: str
    agent_id: str | None
    policy_id: str | None
    policy_name: str
    score_pct: float
    total_checks: int
    passed: int
    failed: int
    not_applicable: int
    controls: list
    ai_summary: str
    ai_recommendations: list
    framework: str
    framework_version: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class ComplianceReportSummary(BaseModel):
    id: UUID
    tenant_id: UUID
    server_id: UUID
    source: str
    policy_name: str
    score_pct: float
    total_checks: int
    passed: int
    failed: int
    not_applicable: int
    framework: str
    created_at: datetime

    model_config = {"from_attributes": True}


class ComplianceGenerateRequest(BaseModel):
    server_id: UUID
    policy_id: str | None = None
    force_source: str | None = None  # "wazuh" | "ssh" | None (auto)


class ComplianceRemediateRequest(BaseModel):
    recommendation_index: int | None = None  # usado quando mode="recommendations"
    mode: str = "recommendations"  # "recommendations" | "controls"


# ── Governance / Trust Score schemas ─────────────────────────────────────────

class TrustScoreRead(BaseModel):
    id: UUID
    tenant_id: UUID
    framework: str
    score_pct: float
    breakdown: dict
    narrative: str
    computed_at: datetime

    model_config = {"from_attributes": True}


class FrameworkScoreItem(BaseModel):
    framework: str
    score_pct: float
    computed_at: datetime

    model_config = {"from_attributes": True}


class GovernanceSummary(BaseModel):
    eternity_score: float | None
    cis_score: float | None
    nist_score: float | None
    iso_score: float | None
    narrative: str
    computed_at: datetime | None
    scores: list[TrustScoreRead]
